pkg/blackduck/ctl_blackduck.go

```diff
@@ -443,71 +311,29 @@ func (ctl *CRSpecBuilderFromCobraFlags) SetCRSpecFieldByFlag(f *pflag.Flag) {
-			ctl.blackDuckSpec.CertificateName = ctl.CertificateName
-		case "certificate-file-path":
-			data, err := util.ReadFileData(ctl.CertificateFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
-			}
-			ctl.blackDuckSpec.Certificate = data
-		case "certificate-key-file-path":
-			data, err := util.ReadFileData(ctl.CertificateKeyFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
-			}
-			ctl.blackDuckSpec.CertificateKey = data
-		case "proxy-certificate-file-path":
-			data, err := util.ReadFileData(ctl.ProxyCertificateFilePath)
-			if err != nil {
-				log.Fatalf("failed to read certificate file: %+v", err)
-			}
-			ctl.blackDuckSpec.ProxyCertificate = data
-		case "auth-custom-ca-file-path":
-			data, err := util.ReadFileData(ctl.AuthCustomCAFilePath)
-			if err != nil {
-				log.Fatalf("failed to read authCustomCA file: %+v", err)
-			}
-			ctl.blackDuckSpec.AuthCustomCA = data
-		case "type":
-			ctl.blackDuckSpec.Type = ctl.Type
-		case "desired-state":
-			ctl.blackDuckSpec.DesiredState = ctl.DesiredState
+		case "postgres-claim-size":
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "claimSize"}, ctl.PostgresClaimSize)
 		case "migration-mode":
-			if ctl.MigrationMode {
-				ctl.blackDuckSpec.DesiredState = "DbMigrate"
-			}
-		case "image-registries":
-			ctl.blackDuckSpec.ImageRegistries = ctl.ImageRegistries
-		case "license-key":
-			ctl.blackDuckSpec.LicenseKey = ctl.LicenseKey
+			util.SetHelmValueInMap(ctl.args, []string{"status"}, "DbMigrate")
+		//case "image-registries":
+		//
+		//	ctl.blackDuckSpec.ImageRegistries = ctl.ImageRegistries
+
 		case "admin-password":
-			ctl.blackDuckSpec.AdminPassword = util.Base64Encode([]byte(ctl.AdminPassword))
-		case "postgres-password":
-			ctl.blackDuckSpec.PostgresPassword = util.Base64Encode([]byte(ctl.PostgresPassword))
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "adminPassword"}, ctl.AdminPassword)
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "isExternal"}, false)
 		case "user-password":
-			ctl.blackDuckSpec.UserPassword = util.Base64Encode([]byte(ctl.UserPassword))
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "userPassword"}, ctl.UserPassword)
+			util.SetHelmValueInMap(ctl.args, []string{"postgres", "isExternal"}, false)
 		case "registry":
-			if ctl.blackDuckSpec.RegistryConfiguration == nil {
-				ctl.blackDuckSpec.RegistryConfiguration = &api.RegistryConfiguration{}
-			}
-			ctl.blackDuckSpec.RegistryConfiguration.Registry = ctl.Registry
+			util.SetHelmValueInMap(ctl.args, []string{"registry"}, ctl.Registry)
 		case "pull-secret-name":
-			if ctl.blackDuckSpec.RegistryConfiguration == nil {
-				ctl.blackDuckSpec.RegistryConfiguration = &api.RegistryConfiguration{}
-			}
-			ctl.blackDuckSpec.RegistryConfiguration.PullSecrets = ctl.PullSecrets
+			util.SetHelmValueInMap(ctl.args, []string{"imagePullSecrets"}, ctl.PullSecrets)
```

Comment:
So I think we are going to have to change this to create a list of maps
```
 imagePullSecrets
 - name: registryName
```
Maybe like this...
```
pullSecretList := []map[string]interface{}{}
for loop ctl.imagePullSecrets {
    key, val := imagePullSecret.Split(":") // How are imagePullSecrets formatted?
    pullSecretList = append(pullSecretList, map[string]interface{}{key: val}
}
util.SetHelmValueInMap(ctl.args, []string{"imagePullSecrets"}, pullSecretList)
```

Reply:
done

Reply:
Ah cool nice!